reference/shaders-hlsl/vert/basic.vert

```diff
@@ -0,0 +1,47 @@
+struct _UBO
+{
+    float4x4 uMVP;
+};
+
+cbuffer UBO
+{
+    _UBO _16;
+};
+uniform float4 gl_HalfPixel;
+
+static float4 gl_Position;
+static float4 aVertex;
+static float3 vNormal;
+static float3 aNormal;
+
+struct InputVert
+{
+    float3 aNormal : TEXCOORD0;
+    float4 aVertex : TEXCOORD1;
+};
+
+struct OutputVert
+{
+    float3 vNormal : TEXCOORD0;
+    float4 gl_Position : POSITION;
+};
+
+void vert_main()
+{
+    gl_Position = mul(transpose(_16.uMVP), aVertex);
```

Comment:
The other way is to unroll the multiply, but that leads to less clean though.

```
M * V = M[0] * V.x + M[1] * V.y + M[2] * V.z + M[3] * V.w
```
If the underlying matrix is column-major. HLSL is row-major, so it should just work reverse the multiplication order if that's doable in HLSL.